Refactor using tqdm and pathlib

CiSong10 opened the pull request:
Rewrite some functions using tqdm process bar and pathlib. Make other small code improvements as well.

PatBall1 commented:
Happy new year @CiSong10 ! And thank you for your contributions. Apologies, but I am overloaded at the moment and am not able to review right now but will aim to do so in a week or so.

CiSong10 commented:
Thank you for the AI-generated review. I’ve made the requested changes and added my responses to the comments below. Please let me know if it's ok or I need to make further changes.

## Response to "Issues & Concerns"
1. My bad. Corrected it.
2. No action required. 
3. `cv2.RETR_TREE` → `cv2.RETR_EXTERNAL`. I think `RETR_EXTERNAL` is better because it can avoid holes or nested tree canopy contours, which are not ideal output. It should also be slightly more time-efficient than `RETR_TREE`. That said, I’m happy to switch back if you prefer.
4. No action required
5. This line is not used in the function. I commented it out in case it’s needed in the future.
6. Agree. I found this inconsistency in the `clean_predictions()` function and refactored it using `pathlib`.
7. I chose not to log a warning to avoid overly noisy output, but I can add one if you think it’s useful.

## Response to "Potential Bugs":
1. Apologies for my oversight. Fixed it.
2. No action required. 

## Response to "Suggestions"
1. No action required
2. No action required
3. Made a few updates to the docstring.
